### PR TITLE
[Compiler] Version Fix for Window Builds

### LINKED ISSUE #31745: Build fails on windows when there's no system-provided python
`getVersion` function uses `.exe` suffix when running on windows:

https://github.com/dart-lang/sdk/blob/6d5f2f09aa9fb70b71dace1e3d6163c4f6f372c7/utils/compiler/create_snapshot_entry.dart#L9-L12

However, `depot_tools` root (which should be in PATH) contains only `python.bat`. Because of this, system-provided `python.exe` will be used (if it exists).

On systems without other python installations, build process fails:
```
ninja -C out\ReleaseX64 create_sdk
ninja: Entering directory `out\ReleaseX64'

FAILED: gen/utils/compiler/utils_wrapper.dart gen/utils/compiler/dart2js.dart
c:/depot_tools/win_tools-2_7_6_bin/python/bin/python.exe ../../build/gn_run_binary.py compiled_action dart.exe --packages=C:/dart-sdk/sdk/.packages C:/dart-sdk/sdk/utils/compiler/create_snapshot_entry.dart --output_dir=C:/dart-sdk/sdk/out/ReleaseX64/gen/utils/compiler --dart2js_main=pkg/compiler/lib/src/dart2js.dart
Command failed: ./dart.exe --packages=C:/dart-sdk/sdk/.packages C:/dart-sdk/sdk/utils/compiler/create_snapshot_entry.dart --output_dir=C:/dart-sdk/sdk/out/ReleaseX64/gen/utils/compiler --dart2js_main=pkg/compiler/lib/src/dart2js.dart
output: Unhandled exception:
ProcessException: The system cannot find the file specified.

  Command: python.exe C:\dart-sdk\sdk\tools\print_version.py
#0      _rootHandleUncaughtError.<anonymous closure> (dart:async/zone.dart:1114)
#1      _microtaskLoop (dart:async/schedule_microtask.dart:41)
#2      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#3      _Timer._runTimers (dart:isolate-patch/dart:isolate/timer_impl.dart:376)
#4      _Timer._handleMessage (dart:isolate-patch/dart:isolate/timer_impl.dart:401)
#5      _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:163)


ninja: build stopped: subcommand failed.
BUILD FAILED
```

The simplest fix is to change `.exe` to `.bat` in the script above.

### LINKED ISSUE #32036: Issues Building on Windows
Hello,

The [wiki on building the sdk](https://github.com/dart-lang/sdk/wiki/Building) is missing some dependencies for Windows:

1. "Desktop development with C++" Visual Studio workload - Visual Studio by itself is not enough
1. Windows 10 SDK
1. Python 2.7 - Chromium's depot tools provides "python.bat", but the build requires "python.exe" in the path.

### PR DESCRIPTION
**Pull request by MichealReed**
Closes https://github.com/dart-lang/sdk/issues/31745
Closes https://github.com/dart-lang/sdk/issues/32036

### CODE REVIEW

**Comment by MichealReed** [comment]
@googlebot I signed it!

File: utils/compiler/create_snapshot_entry.dart [hunk 1]
```diff
@@ -10,7 +10,7 @@ import 'dart:io';
 import 'dart:async';
 
 Future<String> getVersion(var rootPath) {
-  var suffix = Platform.operatingSystem == 'windows' ? '.exe' : '';
+  var suffix = Platform.operatingSystem == 'windows' ? '.bat' : '';
   var printVersionScript = rootPath.resolve("tools/make_version.py");
   return Process.run(
           "python$suffix", [printVersionScript.toFilePath(), "--quiet"])
```
Comment (sigmundch) [inline utils/compiler/create_snapshot_entry.dart:16]: @johnniwinther sent me the following suggestion:

> I'd add `runInShell: true` here instead and leave out the '$suffix' part. That'll let Windows find the python version that it normally finds. On my setup I actually get different versions depending on the suffix and my default is the .exe version.

Would you mind making that change?

**Review by sigmundch — COMMENTED** [review]
Thanks for the PR @MichealReed !

I asked @johnniwinther to take a look, since he also develops on Windows. 

I'm not sure if you saw the conversation in the gerrit CL that was generated from your PR, so I'm copying the comment here.

One idea we discussed there to change how we get this version by updating the GN rules instead of invoking a python script here. That will be a larger change, so I'm happy to adopt your PR first and for us to up with that change separately.

**Comment by MichealReed** [comment]
@sigmundch glad this could spawn good conversation. Should be all set here, I changed it to use runInShell.

**Comment by sigmundch** [comment]
Awesome, thanks! I'll trigger the submit from the mirrored gerrit CL

**Pull request closed without merging**